libraries-3/src/test/java/com/baeldung/jnats/NatsClientLiveTest.java

```diff
@@ -1,115 +1,128 @@
+            assertEquals("hello back", new String(message.getData()));
+        }
     }
 
     @Test
-    public void whenWildCardSubscription_andNotMatchTopic_NoMessageReceived() throws Exception {
-
-        NatsClient client = connectClient();
-
-        SyncSubscription fooSubscription = client.subscribeSync("foo.*");
-
-        client.publishMessage("foo.bar.plop", "bar.foo", "hello there");
-
-        Message message = fooSubscription.nextMessage(200);
-        assertNull("Got message!", message);
-
-
-        SyncSubscription barSubscription = client.subscribeSync("foo.>");
+    public void whenWildCardSubscription_andMatchTopic_MessageReceived() throws Exception {
+        try (NatsClient client = connectClient()) {
+            Subscription fooStarSubscription = client.subscribeSync("foo.*");
 
-        client.publishMessage("foo.bar.plop", "bar.foo", "hello there");
+            client.publishMessage("foo.star", "hello foo star");
 
-        message = barSubscription.nextMessage(200);
-        assertNotNull("No message!", message);
-        assertEquals("hello there", new String(message.getData()));
+            Message message = fooStarSubscription.nextMessage(TIMEOUT_MILLIS);
+            assertNotNull(message, "No message!");
+            assertEquals("hello foo star", new String(message.getData()));
 
+            Subscription fooGreaterSubscription = client.subscribeSync("foo.>");
+            client.publishMessage("foo.greater.than", "hello foo greater");
 
+            message = fooGreaterSubscription.nextMessage(TIMEOUT_MILLIS);
+            assertNotNull(message, "No message!");
+            assertEquals("hello foo greater", new String(message.getData()));
+        }
     }
 
-
     @Test
-    public void givenRequest_ReplyReceived() {
+    public void whenWildCardSubscription_andNotMatchTopic_NoMessageReceived() throws Exception {
+        try (NatsClient client = connectClient()) {
+            Subscription starSubscription = client.subscribeSync("foo.*");
+
+            client.publishMessage("foo.bar.plop", "hello there");
 
-        NatsClient client = connectClient();
-        client.installReply("salary.requests", "denied!");
+            Message message = starSubscription.nextMessage(TIMEOUT_MILLIS);
+            assertNull(message, "Got message!");
 
-        Message reply = client.makeRequest("salary.requests", "I need a raise.");
-        assertNotNull("No message!", reply);
-        assertEquals("denied!", new String(reply.getData()));
+            Subscription greaterSubscription = client.subscribeSync("foo.>");
+            client.publishMessage("foo.bar.plop", "hello there");
 
+            message = greaterSubscription.nextMessage(TIMEOUT_MILLIS);
+            assertNotNull(message, "No message!");
+            assertEquals("hello there", new String(message.getData()));
+        }
     }
 
     @Test
     public void givenQueueMessage_OnlyOneReceived() throws Exception {
```

Comment:
Test names still not matching BDD syntax

Reply:
This test and name was unchanged, so none of this is on me. I will fix though.